units/check_blocked_tasks.go

```diff
@@ -84,29 +96,29 @@ func (j *checkBlockedTasksJob) Run(ctx context.Context) {
 			"job":                 j.ID(),
 			"source":              checkBlockedTasks,
 		})
-		return
+		return nil
 	}
 
 	tasksToCheck, err := task.Find(task.ByIds(taskIds))
 	if err != nil {
 		j.AddError(errors.Wrapf(err, "getting tasks to check in distro '%s'", j.DistroId))
-		return
+		return nil
 	}
+	return tasksToCheck
+}
 
-	dependencyCache := map[string]task.Task{}
-	numTasksModified := 0
-	numChecksThatUpdatedTasks := 0
-	for _, t := range tasksToCheck {
-		numModified, err := checkUnmarkedBlockingTasks(&t, dependencyCache)
-		j.AddError(err)
-		numTasksModified += numModified
-		if numTasksModified > 0 {
-			numChecksThatUpdatedTasks++
-		}
+func (j *checkBlockedTasksJob) getTasksToCheckContainer() []task.Task {
+	query := task.IsUndispatchedContainerTasksQuery()
+	query[task.ContainerAllocatedKey] = false
+	tasksToCheck, err := task.FindAll(db.Query(query).Sort([]string{task.ActivatedTimeKey}))
+	if err != nil {
+		j.AddError(errors.Wrap(err, "getting container tasks to check"))
+		return nil
 	}
+	return tasksToCheck
 }
 
-func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.Task) (int, error) {
+func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.Task) error {
 	catcher := grip.NewBasicCatcher()
 
 	dependenciesMet, err := t.DependenciesMet(dependencyCaches)
```

Comment:
Is it necessary to double-check `DependenciesMet` here? I imagine the query itself for tasks could filter out tasks whose dependencie are already met.

I'm also fine if we just want to make a separate ticket to investigate this later.

Reply:
Part of https://jira.mongodb.org/browse/EVG-20855